VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs

```diff
@@ -98,14 +98,28 @@ public void DoImport(Stream backupStream, Action<ExportImportProgressInfo> progr
                             {
                                 reader.Read();
 
+                                var inventoryInfoChunk = new List<InventoryInfo>();
+
                                 while (reader.TokenType != JsonToken.EndArray)
                                 {
                                     var inventoryInfo = AbstractTypeFactory<InventoryInfo>.TryCreateInstance();
 
                                     inventoryInfo = _jsonSerializer.Deserialize(reader, inventoryInfo.GetType()) as InventoryInfo;
-                                    inventoryInfos.Add(inventoryInfo);
+
+                                    inventoryInfoChunk.Add(inventoryInfo);
 
                                     reader.Read();
+
+                                    if (inventoryInfoChunk.Count >= BatchSize || reader.TokenType == JsonToken.EndArray)
+                                    {
+                                        progressInfo.ProcessedCount += inventoryInfoChunk.Count;
+                                        progressInfo.Description = $"{progressInfo.ProcessedCount} inventories records have been imported";
```

Comment:
`records have been imported` means that the progress should be updated _after_ importing.

Reply:
Changed.